### PR TITLE
Update `overlays/additional` directory for the upcoming Release (Python 3.12, Improved Naming)

### PR DESCRIPTION
**Pull request by atheo89**
Related to: https://issues.redhat.com/browse/RHOAIENG-28585

## Description
This PR updates the ODH Tier 1 notebook imagestreams in the `overlays/additional` directory to prepare for the upcoming release, based on Python 3.12. The changes include:

- Almost all the workbenches and Runtimes  **except** ROCm TensorFlow 
- Standardize the naming pattern on the UI for example the cuda minimal in the past had the name:
`CUDA` and now it is `Jupyter Minimal CUDA Python 3.12`  the same pattern follow to the rest.
This is a small porpotion of how they look like on the UI:
Screenshot 2025-07-09 at 13 26 41

- This how the new set of runtime images look on UI:
Screenshot 2025-07-09 at 13 21 13

Screen recording: https://drive.google.com/file/d/1T8LOBa6x8hAptcZ7re5YZP4dapqtHNoW/view?usp=drive_link

**NOTEs:**
1. We don't include the ROCm TensorFlow due to this: https://github.com/opendatahub-io/notebooks/pull/1259#issuecomment-3031791448
2. The `params-latest.env` file includes image builds from `ghrc` [link](https://github.com/opendatahub-io/notebooks/pull/1259#issuecomment-3031791448), this should change once [this](https://issues.redhat.com/browse/RHOAIENG-28826) or [this](https://issues.redhat.com/browse/RHOAIENG-28188) is in place. **Will follow up PR maybe on the code freeze.** 

## How Has This Been Tested?
1. Run locally `kustomize build manifests/overlays/additional > output.yaml` it should be executed without errors/issues.
2. On cluster use DevFlag on workbench section and inspect the new set of images on the DSP
```
    workbenches:
      devFlags:
        manifests:
          - contextDir: manifests
            sourcePath: ''
            uri: 'https://github.com/atheo89/notebooks/archive/manifests-28585.tar.gz'
      managementState: Managed
      workbenchNamespace: rhods-notebooks
```

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

## Summary by CodeRabbit

* **New Features**
  * Introduced multiple new beta notebook images based on Python 3.12, including minimal, datascience, PyTorch (CUDA and ROCm), TensorFlow, TrustyAI, and code-server variants with CPU and GPU support.
  * Added new runtime images for Elyra pipeline execution with Python 3.12, covering minimal, datascience, PyTorch, TensorFlow, and ROCm variants.
  * Included detailed metadata and software stack information for all new images to improve discoverability and usage.

* **Documentation**
  * Updated README to list new beta notebook images and note their Python 3.12 and 2025.b beta status.

* **Chores**
  * Extended configuration files to reference new Python 3.12 images and corresponding commit hashes.
  * Enhanced parameterization of image references and build commit annotations for easier future updates.

### CODE REVIEW

**Comment by atheo89** [comment]
CI fails due to unauthorized access, these images will be updated once the builds are in place.
```
time="2025-07-09T09:18:49Z" level=fatal msg="Error parsing image name \"docker://ghcr.io/opendatahub-io/notebooks/workbench-images@sha256:3eaabc3d87c152e1c9255ee5eecd29812d0da76017a2474aa0c59a06382ecc9a\": unable to retrieve auth token: invalid username/password: unauthorized"
Couldn't download image config metadata with skopeo tool!
```

**Comment by jiridanek** [comment]
Don't see what's the advantage of using `overlays/additional` for this. Hopefully the effort to make CI work with this manifest location won't be too much.

/lgtm

**Comment by atheo89** [comment]
Thanks for youe review folks! i will approve this in order to start checking the builds (hopefully tommrow)
/approve
